File: gui/orchestra-gui/js/pages/onboarding.js
// Orchestra — First-run onboarding wizard
(function () {
  const { icons } = window;
  const API = window.ORCH_API || 'http://localhost:3000';
  const STORAGE_KEY = 'orchestraOnboarded';

  let step = 1; // 1 = Welcome, 2 = API Key, 3 = Try it
  let apiKeyValue = '';
  let apiSaving = false;
  let apiSaved = false;

  function escapeHTML(s) {
    return String(s).replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
  }

  // ── Check whether to show ──────────────────────────────────────────────────
  function check() {
    if (localStorage.getItem(STORAGE_KEY)) return;
    show();
  }

  function markDone() {
    localStorage.setItem(STORAGE_KEY, '1');
  }

  // ── Render ─────────────────────────────────────────────────────────────────
  function renderStep() {
    switch (step) {
      case 1: return renderWelcome();
      case 2: return renderApiKey();
      case 3: return renderTryIt();
      default: return renderWelcome();
    }
  }

  function renderProgressDots() {
    return `
      <div class="ob-dots">
        ${[1, 2, 3].map(n => `<div class="ob-dot ${step === n ? 'is-active' : step > n ? 'is-done' : ''}"></div>`).join('')}
      </div>`;
  }

  function renderWelcome() {
    return `
      <div class="ob-step">
        <div class="ob-icon" style="background:linear-gradient(135deg,var(--accent),var(--teal))">
          ${icons.logo ? icons.logo(32) : icons.sparkles(32)}
        </div>
        <h2 class="ob-title">Welcome to Orchestra</h2>
        <p class="ob-body">Orchestra is an AI task runner that coordinates agents across your tools, files, and APIs — so you can delegate entire workflows, not just individual questions.</p>
        <ul class="ob-features">
          <li>${icons.check(13)} <span>Real-time streaming with tool visibility</span></li>
          <li>${icons.check(13)} <span>Multi-step agent coordination (A→E architectures)</span></li>
          <li>${icons.check(13)} <span>Background jobs with live notifications</span></li>
          <li>${icons.check(13)} <span>Integrates with GitHub, Slack, Google Drive, and more</span></li>
        </ul>
        <div class="ob-actions">
          <button class="btn btn--subtle btn--sm" data-ob-skip>Skip setup</button>
          <button class="btn btn--primary" data-ob-next>Get started →</button>
        </div>
      </div>`;
  }

  function renderApiKey() {
    return `
      <div class="ob-step">
        <div class="ob-icon" style="background:var(--accent-dim);color:var(--accent)">
          ${icons.key(28)}
        </div>
        <h2 class="ob-title">Connect an AI provider</h2>
        <p class="ob-body">Orchestra needs an API key to send tasks to a language model. Your Anthropic key is the easiest way to get started.</p>

        <div class="ob-field">
          <label>Anthropic API key</label>
          <input
            class="input"
            type="password"
            placeholder="sk-ant-…"
            value="${escapeHTML(apiKeyValue)}"
            data-ob-key-input
            autocomplete="off"
            style="font-family:var(--font-mono)"
          />
          <div style="font-size:12px;color:var(--text-3);margin-top:6px">
            Get your key at <strong>console.anthropic.com</strong> → API Keys.
            It is stored server-side and never sent to the browser.
          </div>
        </div>

        ${apiSaved ? `
          <div style="display:flex;align-items:center;gap:8px;margin-top:12px;color:var(--success);font-size:13.5px">
            ${icons.check(14)} Key saved successfully
          </div>` : ''}

        <div class="ob-actions">
          <button class="btn btn--ghost" data-ob-back>← Back</button>
          <div style="display:flex;gap:8px">
            <button class="btn btn--subtle btn--sm" data-ob-skip-key>Skip for now</button>
            <button
              class="btn btn--primary"
              data-ob-save-key
              ${apiSaving ? 'disabled' : ''}
            >${apiSaving ? 'Saving…' : apiSaved ? 'Next →' : 'Save &amp; continue'}</button>
          </div>
        </div>
      </div>`;
  }

  function renderTryIt() {
    const prompts = [
      { title: 'Research task',   text: 'Summarize the latest AI safety papers from the last 30 days.' },
      { title: 'Code task',       text: 'Write a Python script that monitors a folder and logs new files to a CSV.' },
      { title: 'Analysis task',   text: 'Analyze the attached CSV and produce a summary with key trends.' },
      { title: 'Automation task', text: 'Check my GitHub repo for open issues older than 14 days and draft responses.' },
    ];

    return `
      <div class="ob-step">
        <div class="ob-icon" style="background:var(--success-dim);color:var(--success)">
          ${icons.sparkles(28)}
        </div>
        <h2 class="ob-title">You're ready. Try a task.</h2>
        <p class="ob-body">Pick a template below or jump straight to the chat. Orchestra will plan, use tools, and show you every step as it works.</p>

        <div class="ob-prompts">
          ${prompts.map(p => `
            <button class="ob-prompt" data-ob-prompt="${escapeHTML(p.text)}">
              <div class="ob-prompt__title">${escapeHTML(p.title)}</div>
              <div class="ob-prompt__text">${escapeHTML(p.text)}</div>
            </button>`).join('')}
        </div>

        <div class="ob-actions" style="margin-top:24px">
          <button class="btn btn--ghost" data-ob-back>← Back</button>
          <button class="btn btn--primary" data-ob-finish>Open chat →</button>
        </div>
      </div>`;
  }

  // ── DOM ─────────────────────────────────────────────────────────────────────
  function getOverlay() { return document.getElementById('onboarding-overlay'); }
  function getPanel()   { return document.getElementById('onboarding-panel'); }

  function repaint() {
    const panel = getPanel();
    if (!panel) return;
    panel.innerHTML = renderProgressDots() + renderStep();
    wireStep();
  }

  function show() {
    const overlay = getOverlay();
    if (!overlay) return;
    overlay.classList.add('is-open');
    repaint();
  }

  function close(navigateTo) {
    const overlay = getOverlay();
    if (overlay) overlay.classList.remove('is-open');
    markDone();
    if (navigateTo) location.hash = navigateTo;
  }

  async function doSaveKey() {
    const val = apiKeyValue.trim();
    if (!val) {
      step = 3;
      repaint();
      return;
    }
    apiSaving = true;
    repaint();
    try {
      const r = await fetch(`${API}/v1/config/keys`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ anthropic_api_key: val }),
        signal: AbortSignal.timeout(8000),
      });
      if (!r.ok) throw new Error(`HTTP ${r.status}`);
      apiSaved = true;
      if (window.Orchestra?.toast) window.Orchestra.toast.show('Anthropic key saved', 'success', 3000);
      setTimeout(() => { step = 3; repaint(); }, 600);
    } catch (e) {
      if (window.Orchestra?.toast) window.Orchestra.toast.show('Failed to save key — you can add it later in Settings', 'error', 5000);
      step = 3;
    } finally {
      apiSaving = false;
      repaint();
    }
  }

  function wireStep() {
    const panel = getPanel();
    if (!panel) return;

    const keyInput = panel.querySelector('[data-ob-key-input]');
    if (keyInput) {
      keyInput.addEventListener('input', (e) => { apiKeyValue = e.target.value; });
      keyInput.addEventListener('keydown', (e) => { if (e.key === 'Enter') doSaveKey(); });
    }

    panel.querySelector('[data-ob-next]')?.addEventListener('click', () => { step++; repaint(); });
    panel.querySelector('[data-ob-back]')?.addEventListener('click', () => { step = Math.max(1, step - 1); repaint(); });
    panel.querySelector('[data-ob-skip]')?.addEventListener('click', () => close('#/chat'));
    panel.querySelector('[data-ob-skip-key]')?.addEventListener('click', () => { step = 3; repaint(); });
    panel.querySelector('[data-ob-save-key]')?.addEventListener('click', doSaveKey);
    panel.querySelector('[data-ob-finish]')?.addEventListener('click', () => close('#/chat'));

    panel.querySelectorAll('[data-ob-prompt]').forEach(btn => {
      btn.addEventListener('click', () => {
        close('#/chat');
        // Pre-fill composer after navigation
        setTimeout(() => {
          const ta = document.querySelector('[data-composer]');
          if (ta) {
            ta.value = btn.dataset.obPrompt;
            ta.dispatchEvent(new Event('input'));
            ta.focus();
          }
        }, 150);
      });
    });
  }

  // ── Close on overlay click ─────────────────────────────────────────────────
  document.addEventListener('click', (e) => {
    const overlay = getOverlay();
    if (overlay && e.target === overlay) close();
  });

  window.Orchestra = window.Orchestra || {};
  window.Orchestra.onboarding = { check, show };
})();

File: orchestra/code_agent/ui/onboarding.py
ONBOARDING_HTML = """<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Welcome to Orchestra</title>
<style>
*,*::before,*::after{box-sizing:border-box;margin:0;padding:0}
:root{
  --bg:#0d1117;--bg2:#161b22;--bg3:#21262d;
  --border:#30363d;--text:#e6edf3;--muted:#8b949e;
  --blue:#58a6ff;--green:#3fb950;--purple:#a78bfa;
  --red:#f85149;--orange:#f0883e;
}
html,body{height:100%;overflow:hidden}
body{background:var(--bg);color:var(--text);font-family:-apple-system,BlinkMacSystemFont,'Segoe UI',sans-serif;display:flex;align-items:center;justify-content:center}

/* Outer shell */
.shell{width:100%;max-width:600px;padding:24px;height:100%;display:flex;flex-direction:column;overflow-y:auto}
@media(max-width:480px){.shell{padding:16px}}

/* Progress bar */
.progress{display:flex;gap:6px;margin-bottom:36px;justify-content:center}
.step-dot{width:8px;height:8px;border-radius:50%;background:var(--border);transition:background .3s}
.step-dot.active{background:var(--blue)}
.step-dot.done{background:var(--green)}

/* Step card */
.step{display:none;flex:1;flex-direction:column;justify-content:center}
.step.active{display:flex;animation:fadeUp .3s ease}
@keyframes fadeUp{from{opacity:0;transform:translateY(12px)}to{opacity:1;transform:none}}

.step-label{font-size:11px;font-weight:700;color:var(--muted);text-transform:uppercase;letter-spacing:.8px;margin-bottom:10px}
.step-title{font-size:26px;font-weight:800;margin-bottom:8px;line-height:1.2}
.step-sub{font-size:14px;color:var(--muted);margin-bottom:28px;line-height:1.6}

/* Vertical picker cards */
.vertical-grid{display:grid;grid-template-columns:1fr 1fr;gap:12px;margin-bottom:24px}
@media(max-width:420px){.vertical-grid{grid-template-columns:1fr}}
.vertical-card{
  border:2px solid var(--border);border-radius:12px;padding:16px;cursor:pointer;
  transition:all .2s;background:var(--bg2);
}
.vertical-card:hover{border-color:rgba(88,166,255,.4);background:rgba(88,166,255,.04)}
.vertical-card.selected{border-color:var(--blue);background:rgba(88,166,255,.08)}
.vertical-card.selected .vc-icon{filter:none}
.vc-icon{font-size:28px;margin-bottom:8px;filter:grayscale(.6)}
.vc-label{font-size:14px;font-weight:700;margin-bottom:2px}
.vc-desc{font-size:12px;color:var(--muted);line-height:1.4}

/* Input styles */
.field-label{font-size:12px;color:var(--muted);margin-bottom:6px;display:block}
.field-input{width:100%;background:var(--bg);border:1px solid var(--border);border-radius:8px;padding:10px 14px;color:var(--text);font-size:14px;font-family:inherit;outline:none;transition:border-color .15s;margin-bottom:16px}
.field-input:focus{border-color:var(--blue)}
.field-input::placeholder{color:#484f58}
.field-group{position:relative}
.field-group .field-input{padding-right:40px}
.field-toggle{position:absolute;right:12px;top:50%;transform:translateY(-50%);background:none;border:none;color:var(--muted);cursor:pointer;font-size:13px}

/* API key grid */
.api-keys{display:flex;flex-direction:column;gap:12px;margin-bottom:8px}
.api-key-row{display:flex;flex-direction:column;gap:4px}
.api-key-row .provider-badge{font-size:11px;font-weight:700;color:var(--muted);text-transform:uppercase;letter-spacing:.5px}

/* Action buttons */
.btn-row{display:flex;gap:10px;margin-top:auto;padding-top:20px}
.btn{padding:11px 22px;border-radius:8px;font-size:14px;font-weight:600;cursor:pointer;font-family:inherit;border:none;transition:opacity .15s;min-height:44px}
.btn-primary{background:var(--blue);color:#fff;flex:1}.btn-primary:hover{opacity:.85}
.btn-ghost{background:transparent;color:var(--muted);border:1px solid var(--border)}.btn-ghost:hover{color:var(--text)}
.btn-success{background:var(--green);color:#000;flex:1}

/* Skip link */
.skip{display:block;text-align:center;margin-top:12px;font-size:12px;color:var(--muted);cursor:pointer;min-height:44px;line-height:44px}
.skip:hover{color:var(--text)}

/* Checkmark animation */
.check-anim{display:flex;align-items:center;justify-content:center;width:64px;height:64px;border-radius:50%;background:rgba(63,185,80,.12);margin:0 auto 24px}
.check-anim svg{width:36px;height:36px}

/* Summary list */
.summary{background:var(--bg2);border:1px solid var(--border);border-radius:10px;padding:16px;margin-bottom:20px}
.summary-row{display:flex;align-items:center;gap:10px;padding:6px 0;font-size:13px;border-bottom:1px solid rgba(48,54,61,.5)}
.summary-row:last-child{border-bottom:none}
.summary-key{color:var(--muted);width:110px;flex-shrink:0}
.summary-val{font-weight:600;flex:1}
.summary-val.ok{color:var(--green)}
.summary-val.skip{color:var(--muted)}
</style>
</head>
<body>
<div class="shell">
  <!-- Progress dots -->
  <div class="progress">
    <div class="step-dot active" id="dot0"></div>
    <div class="step-dot" id="dot1"></div>
    <div class="step-dot" id="dot2"></div>
    <div class="step-dot" id="dot3"></div>
    <div class="step-dot" id="dot4"></div>
  </div>

  <!-- Step 0: Welcome -->
  <div class="step active" id="step0">
    <div class="step-label">Welcome</div>
    <div class="step-title">Let's get you set up</div>
    <div class="step-sub">Orchestra is an autonomous AI assistant for healthcare, legal, and finance professionals. This wizard takes about 2 minutes.</div>
    <div class="btn-row">
      <button class="btn btn-primary" onclick="nextStep()">Get started →</button>
    </div>
    <a class="skip" onclick="skipAll()">Skip setup and go to app</a>
  </div>

  <!-- Step 1: Vertical picker -->
  <div class="step" id="step1">
    <div class="step-label">Step 1 of 4</div>
    <div class="step-title">What do you work on?</div>
    <div class="step-sub">Orchestra uses this to tailor prompts, compliance rules, and model selection.</div>
    <div class="vertical-grid" id="verticalGrid">
      <div class="vertical-card" onclick="selectVertical('healthcare', this)">
        <div class="vc-icon">🏥</div>
        <div class="vc-label">Healthcare</div>
        <div class="vc-desc">Clinical notes, HIPAA compliance, patient records, medical research</div>
      </div>
      <div class="vertical-card" onclick="selectVertical('legal', this)">
        <div class="vc-icon">⚖️</div>
        <div class="vc-label">Legal</div>
        <div class="vc-desc">Contract review, client matter files, privilege management, GDPR</div>
      </div>
      <div class="vertical-card" onclick="selectVertical('finance', this)">
        <div class="vc-icon">📈</div>
        <div class="vc-label">Finance</div>
        <div class="vc-desc">LBO models, EDGAR filings, portfolio analysis, quant research</div>
      </div>
      <div class="vertical-card" onclick="selectVertical('engineering', this)">
        <div class="vc-icon">💻</div>
        <div class="vc-label">Engineering</div>
        <div class="vc-desc">Code generation, debugging, architecture, CI/CD automation</div>
      </div>
      <div class="vertical-card" onclick="selectVertical('research', this)">
        <div class="vc-icon">🔬</div>
        <div class="vc-label">Research</div>
        <div class="vc-desc">Literature review, experiments, data analysis, paper writing</div>
      </div>
      <div class="vertical-card" onclick="selectVertical('general', this)">
        <div class="vc-icon">🌐</div>
        <div class="vc-label">General</div>
        <div class="vc-desc">Mixed tasks, writing, planning, and everything else</div>
      </div>
    </div>
    <div class="btn-row">
      <button class="btn btn-ghost" onclick="prevStep()">Back</button>
      <button class="btn btn-primary" onclick="nextStep()">Continue</button>
    </div>
    <a class="skip" onclick="nextStep()">Skip this step</a>
  </div>

  <!-- Step 2: Org name -->
  <div class="step" id="step2">
    <div class="step-label">Step 2 of 4</div>
    <div class="step-title">Name your organization</div>
    <div class="step-sub">This creates your org workspace. You can invite teammates later from the admin panel.</div>
    <label class="field-label">Organization name</label>
    <input id="orgName" class="field-input" placeholder="Acme Legal LLP" maxlength="80">
    <div class="btn-row">
      <button class="btn btn-ghost" onclick="prevStep()">Back</button>
      <button class="btn btn-primary" onclick="nextStep()">Continue</button>
    </div>
    <a class="skip" onclick="nextStep()">Skip — use personal workspace</a>
  </div>

  <!-- Step 3: API keys -->
  <div class="step" id="step3">
    <div class="step-label">Step 3 of 4</div>
    <div class="step-title">Connect your AI providers</div>
    <div class="step-sub">Keys are encrypted and stored server-side. Add at least one to enable AI features. You can add more later in Settings.</div>
    <div class="api-keys">
      <div class="api-key-row">
        <span class="provider-badge">Anthropic</span>
        <div class="field-group">
          <input id="keyAnthropic" class="field-input" type="password" placeholder="sk-ant-…" autocomplete="off">
          <button class="field-toggle" onclick="toggleKey('keyAnthropic', this)">Show</button>
        </div>
      </div>
      <div class="api-key-row">
        <span class="provider-badge">OpenAI</span>
        <div class="field-group">
          <input id="keyOpenAI" class="field-input" type="password" placeholder="sk-…" autocomplete="off">
          <button class="field-toggle" onclick="toggleKey('keyOpenAI', this)">Show</button>
        </div>
      </div>
      <div class="api-key-row">
        <span class="provider-badge">Moonshot (Kimi)</span>
        <div class="field-group">
          <input id="keyMoonshot" class="field-input" type="password" placeholder="sk-…" autocomplete="off">
          <button class="field-toggle" onclick="toggleKey('keyMoonshot', this)">Show</button>
        </div>
      </div>
    </div>
    <div class="btn-row">
      <button class="btn btn-ghost" onclick="prevStep()">Back</button>
      <button class="btn btn-primary" onclick="saveAndNext()">Save keys</button>
    </div>
    <a class="skip" onclick="nextStep()">Skip — I'll add keys later</a>
  </div>

  <!-- Step 4: Done -->
  <div class="step" id="step4">
    <div class="check-anim">
      <svg viewBox="0 0 36 36" fill="none"><circle cx="18" cy="18" r="17" stroke="#3fb950" stroke-width="2"/><path d="M11 18l5 5 9-10" stroke="#3fb950" stroke-width="2.5" stroke-linecap="round" stroke-linejoin="round"/></svg>
    </div>
    <div class="step-label">All set!</div>
    <div class="step-title">Welcome to Orchestra</div>
    <div class="step-sub">Here's what we've configured for you:</div>
    <div class="summary" id="summary"></div>
    <div class="btn-row">
      <button class="btn btn-success" onclick="launchApp()">Open Orchestra →</button>
    </div>
    <a class="skip" onclick="window.location.href='/admin'">Go to admin panel</a>
  </div>
</div>

<script>
const _token = localStorage.getItem('orchestra_token') || '';
let _currentStep = 0;
let _vertical = '';
let _orgCreated = null;
let _keysSaved = [];

const STEPS = 5;

function updateDots() {
  for (let i = 0; i < STEPS; i++) {
    const d = document.getElementById('dot' + i);
    if (i < _currentStep) { d.className = 'step-dot done'; }
    else if (i === _currentStep) { d.className = 'step-dot active'; }
    else { d.className = 'step-dot'; }
  }
}

function showStep(n) {
  document.querySelectorAll('.step').forEach(s => s.classList.remove('active'));
  document.getElementById('step' + n).classList.add('active');
  _currentStep = n;
  updateDots();
}

function nextStep() {
  if (_currentStep < STEPS - 1) showStep(_currentStep + 1);
  if (_currentStep === STEPS - 1) buildSummary();
}
function prevStep() {
  if (_currentStep > 0) showStep(_currentStep - 1);
}

function selectVertical(v, el) {
  _vertical = v;
  document.querySelectorAll('.vertical-card').forEach(c => c.classList.remove('selected'));
  el.classList.add('selected');
}

function toggleKey(id, btn) {
  const input = document.getElementById(id);
  const isText = input.type === 'text';
  input.type = isText ? 'password' : 'text';
  btn.textContent = isText ? 'Show' : 'Hide';
}

async function saveAndNext() {
  const keys = [
    { provider: 'anthropic', value: document.getElementById('keyAnthropic').value.trim(), name: 'ANTHROPIC_API_KEY' },
    { provider: 'openai', value: document.getElementById('keyOpenAI').value.trim(), name: 'OPENAI_API_KEY' },
    { provider: 'moonshot', value: document.getElementById('keyMoonshot').value.trim(), name: 'MOONSHOT_API_KEY' },
  ].filter(k => k.value);

  for (const k of keys) {
    try {
      await fetch('/api/api-keys', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json', 'Authorization': 'Bearer ' + _token },
        body: JSON.stringify({ name: k.name, value: k.value, provider: k.provider }),
      });
      _keysSaved.push(k.provider);
    } catch(e) {}
  }
  nextStep();
}

async function createOrg() {
  const name = document.getElementById('orgName').value.trim();
  if (!name || !_token) return;
  try {
    const r = await fetch('/api/orgs', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json', 'Authorization': 'Bearer ' + _token },
      body: JSON.stringify({ name }),
    });
    if (r.ok) _orgCreated = await r.json();
  } catch(e) {}
}

// Override nextStep for step 2 (org creation)
const _origNext = nextStep;
async function nextStep() {
  if (_currentStep === 2) { await createOrg(); }
  if (_currentStep < STEPS - 1) showStep(_currentStep + 1);
  if (_currentStep === STEPS - 1) buildSummary();
}

function buildSummary() {
  const verticalLabel = {
    healthcare: '🏥 Healthcare', legal: '⚖️ Legal', finance: '📈 Finance',
    engineering: '💻 Engineering', research: '🔬 Research', general: '🌐 General',
  }[_vertical] || 'Not selected';

  const orgRow = _orgCreated
    ? `<div class="summary-row"><span class="summary-key">Organization</span><span class="summary-val ok">${_orgCreated.name}</span></div>`
    : `<div class="summary-row"><span class="summary-key">Organization</span><span class="summary-val skip">Personal workspace</span></div>`;

  const keysRow = _keysSaved.length
    ? `<div class="summary-row"><span class="summary-key">API keys</span><span class="summary-val ok">${_keysSaved.join(', ')} ✓</span></div>`
    : `<div class="summary-row"><span class="summary-key">API keys</span><span class="summary-val skip">None added (local models only)</span></div>`;

  document.getElementById('summary').innerHTML = `
    <div class="summary-row"><span class="summary-key">Vertical</span><span class="summary-val">${verticalLabel}</span></div>
    ${orgRow}
    ${keysRow}
  `;
  // Persist vertical preference
  if (_vertical) localStorage.setItem('orchestra_vertical', _vertical);
}

function launchApp() {
  localStorage.setItem('orchestra_onboarded', '1');
  window.location.href = '/app';
}

function skipAll() {
  localStorage.setItem('orchestra_onboarded', '1');
  window.location.href = '/app';
}

// If already onboarded, go straight to app
if (localStorage.getItem('orchestra_onboarded') === '1') {
  window.location.href = '/app';
}
</script>
</body>
</html>"""
